feat: Add Multi Area Chart

IlyesBenAmara opened the pull request:
# Motivation
This PR adds a new chart : multi area chart for the React/Vue packages. I had to change the `useToggableDomainKey` hook so that it will sort the domain keys based on the max value of each key. The purpose of this is to draw the area with the maximum value first. We will need to add an opacity prop because we might hide a portion of the chart.

![image](https://user-images.githubusercontent.com/22627035/195579291-2d7c6251-0119-4656-9605-ced3f9450652.png)


## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] I have done the work for both React and Vue
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

marrouchi commented on packages/ez-react/src/components/Areas.tsx:
```diff
@@ -0,0 +1,83 @@
+  marker?: MarkerConfig;
+}
+
+export const Areas: FC<AreasProps> = ({
```
```suggestion
export const Area: FC<AreaProps> = ({
```
I think this should be in singular form (also filename).

IlyesBenAmara commented:
- I removed the sorting of the areas, and added an opacity prop for the area.
![image](https://user-images.githubusercontent.com/22627035/196478387-b75bda80-4d76-4b9a-b33e-544aad076049.png)

IlyesBenAmara commented:
I made some changes that were necessary because of the unit tests. One of them is the line color. 
I will bring the necessary changes once I start working on the Vue version :wink:

marrouchi commented on packages/ez-react/src/components/shapes/Area.tsx:
```diff
@@ -4,13 +4,13 @@ import { defaultColor, generateAreaPath } from 'eazychart-core/src';
 }
 
-export const Area: FC<AreaProps> = ({
+export const AreaPath: FC<AreaPathProps> = ({
```
@IlyesBenAmara Let's rename the file as well.

IlyesBenAmara replied:
That is what I intended to do :wink:

marrouchi reviewed (changes requested):
@IlyesBenAmara Looks good to me except it's missing the vue implementation. Can you fork from this branch to create a seperate PR for vue ?

marrouchi commented on packages/ez-react/src/components/shapes/AreaPath.tsx:
```diff
@@ -4,13 +4,13 @@ import { defaultColor, generateAreaPath } from 'eazychart-core/src';
-export const Area: FC<AreaProps> = ({
+export const AreaPath: FC<AreaPathProps> = ({
   shapeData = [],
   curve = 'curveLinear',
```
@IlyesBenAmara let's rename the className here as "ez-area-path"

marrouchi commented on packages/ez-react/src/recipes/area/MultiAreaChart.tsx:
```diff
@@ -0,0 +1,154 @@
+                key={yDomainKey}
+                xDomainKey={xAxis.domainKey}
+                yDomainKey={yDomainKey}
+                area={{
+                  ...area,
+                  opacity: 1 / activeDomainKeys.length,
+                }}
```
```suggestion
                area={area}
```

marrouchi commented on packages/ez-react/src/recipes/area/MultiAreaChart.tsx:
```diff
@@ -0,0 +1,154 @@
+export const MultiAreaChart: FC<MultiAreaChartProps> = ({
+  data,
+  colors = ['#339999', '#993399', '#333399'],
+  area = {
+    stroke: '#339999',
+    strokeWidth: 2,
+    curve: 'curveLinear',
+  },
```
```suggestion
  area = {
    stroke: '#339999',
    strokeWidth: 2,
    curve: 'curveLinear',
    opacity: 0.5
  },
```

IlyesBenAmara commented:
The unit tests do not go through as expected. I made the needed changes into another branch for the Vue version of the multi Area chart.
so in order to repair the unit tests, we need to merge  #88.

marrouchi commented:
We need to add the chart to the list of exported charts in both react & vue.

IlyesBenAmara commented on packages/ez-dev/jest/snapshots/components/shapes/Area.spec.tsx.snap:
```diff
@@ -2,7 +2,7 @@
 
```
This file should definitely be deleted.

IlyesBenAmara commented:
@marrouchi I have addressed the issue of the provided domain Keys and the opacity. You can now review it again.

marrouchi commented on packages/ez-dev/jest/snapshots/recipes/area/AreaChart.spec.tsx.snap:
```diff
@@ -54,7 +55,7 @@ exports[`AreaChart renders an area chart 1`] = `
                     data-testid="ez-point"
                     fill="red"
                     r="2"
-                    stroke="red"
+                    stroke="orange"
```
Why is the fill color different from the stroke ?

marrouchi commented on packages/ez-react/src/recipes/area/AreaChart.stories.tsx:
```diff
@@ -52,3 +61,17 @@ Default.args = {
+  ...defaultArguments,
+  area: { ...defaultArguments.area, opacity: 0.5 },
+  yAxis: {
+    domainKeys: ['yValue2', 'yValue1', 'yValue'],
```
```suggestion
    domainKeys: ['yValue', 'yValue1', 'yValue2'],
```

IlyesBenAmara replied:
Did this on purpose so that the `yValue2` domain key is rendered first. Since we don't sort the domain keys.

marrouchi commented on packages/ez-react/src/recipes/area/MultiAreaChart.tsx:
```diff
@@ -0,0 +1,152 @@
+    domainKey: 'xValue',
+  },
+  yAxis = {
+    domainKeys: ['yValue2', 'yValue1', 'yValue'],
```
```suggestion
    domainKeys: ['yValue', 'yValue1', 'yValue2'],
```
:nail_care:

marrouchi commented on packages/ez-dev/jest/snapshots/recipes/area/AreaChart.spec.tsx.snap:
```diff
@@ -17,9 +17,10 @@ exports[`AreaChart renders an area chart 1`] = `
               <g class="ez-points">
                 <g class="ez-area">
```
The unit test should show two areas instead of a single one.

IlyesBenAmara replied:
This is the area chart

marrouchi commented on packages/ez-vue/tests/unit/recipes/area/MultiAreaChart.spec.tsx:
```diff
@@ -0,0 +1,41 @@
+        },
+        grid: { directions: [] },
+        dimensions,
+        yAxis: {
```
xAxis should be specified here.
